proto/messages.proto

```diff
@@ -206,7 +207,8 @@ message Asset {
 // Actual message is encrypted with AES and sent as additional data
 message External {
   required bytes otr_key = 1;
-  optional bytes sha256 = 2;      // sha256 of ciphertext
+  optional bytes sha256 = 2;      // sha256 of ciphertext, obsolete but required for backward compatibility
```

Comment:
It is true that the enum is redundant: if it has a hash, it has CBC, if it does not, it has GCM.
But we also have a transition period we want to support both and we can have newer clients send "CBC" when they still are in compatibility mode. I think this makes it a little bit better to read but it's not strictly necessary. Other opinions?